docs/src/main/asciidoc/_client-guide/authentication-client.adoc

```diff
@@ -471,6 +471,31 @@ The following example configuration specifies that the default filtering should
 
  <certificate-revocation-list />:: The presence of this element enabled checking the peer's certificate against a certificate revocation list, this element defines both a path to the certificate revocation list and also specifies the maximum number of non-self-issued intermediate certificates that may exist in a certification path
 
+=== Configuring SSLv2Hello
+
+Older JDK versions use ``SSLv2Hello`` during the initial SSL handshake message
+where the SSL/TLS version that will be used for the rest of the communication is
+negotiated.
+
+Using ``SSLv2Hello`` is discouraged, therefore newer JDK versions disable this protocol
+on the client by default. However, they do provide the ability to re-enable it if necessary.
+
+``SSLv2Hello`` can be configured as a supported protocol for the SSL context as follows:
+
+[source,options="nowrap"]
+----
+<protocol names="SSLv2Hello TLSv1">
+----
+
+=== Important to Note:
```

Comment:
I think we could use "*WARNING*" here instead of "=== Important to Note:". I think we should add a sentence at the beginning of this warning that says: "The use of SSLv2Hello is discouraged." just to emphasize it again.

Reply:
I've updated the documentation and also added the missing test scenarios.